### DISCUSSION TITLE
Multithreads support - RuntimeError: can't start new thread

### DISCUSSION THREAD

**Discussion by LijieZhang1998** [pyodide/pyodide#4077] [Q&A]
Hi, I'm porting `pymongo` package to pyodide package. I got an error when I executed my sample code in Node.js. How's the status of threading support in pyodide now? I found one related issue #237. Is there any solution that I can use to work around this error. Thank you.

<details>
  <summary>error</summary>
PythonError: Traceback (most recent call last): File "/lib/python311.zip/_pyodide/_base.py", line 558, in eval_code_async await CodeRunner( File "/lib/python311.zip/_pyodide/_base.py", line 381, in run_async coroutine = eval(self.code, globals, locals) ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ File "", line 10, in File "/lib/python3.11/site-packages/pymongo/mongo_client.py", line 837, in __init__ self._get_topology() File "/lib/python3.11/site-packages/pymongo/mongo_client.py", line 1214, in _get_topology self._topology.open() File "/lib/python3.11/site-packages/pymongo/topology.py", line 192, in open self._ensure_opened() File "/lib/python3.11/site-packages/pymongo/topology.py", line 596, in _ensure_opened self._update_servers() File "/lib/python3.11/site-packages/pymongo/topology.py", line 747, in _update_servers server.open() File "/lib/python3.11/site-packages/pymongo/server.py", line 49, in open self._monitor.open() File "/lib/python3.11/site-packages/pymongo/monitor.py", line 79, in open self._executor.open() File "/lib/python3.11/site-packages/pymongo/periodic_executor.py", line 87, in open thread.start() File "/lib/python311.zip/threading.py", line 957, in start _start_new_thread(self._bootstrap, ()) 
<p><b>RuntimeError: can't start new thread</b></p>
at new_error (/Users/lijie.zhang/workspace/pyodide-repo/dist/pyodide.asm.js:9:12562)
at wasm://wasm/0224d1de:wasm-function[295]:0x158a38
at wasm://wasm/0224d1de:wasm-function[451]:0x15fe7d
at _PyCFunctionWithKeywords_TrampolineCall (/Users/lijie.zhang/workspace/pyodide-repo/dist/pyodide.asm.js:9:119469)
at wasm://wasm/0224d1de:wasm-function[1056]:0x1a321d
at wasm://wasm/0224d1de:wasm-function[3383]:0x28a37c
at wasm://wasm/0224d1de:wasm-function[2036]:0x1e4108
at wasm://wasm/0224d1de:wasm-function[1063]:0x1a3705
at wasm://wasm/0224d1de:wasm-function[1066]:0x1a39c6
at wasm://wasm/0224d1de:wasm-function[1067]:0x1a3a68
at wasm://wasm/0224d1de:wasm-function[3196]:0x268abf
at wasm://wasm/0224d1de:wasm-function[3197]:0x26e8d0
at wasm://wasm/0224d1de:wasm-function[1069]:0x1a3b90
at wasm://wasm/0224d1de:wasm-function[1064]:0x1a3820
at wasm://wasm/0224d1de:wasm-function[439]:0x15f603
at Module.callPyObjectKwargs (/Users/lijie.zhang/workspace/pyodide-repo/dist/pyodide.asm.js:9:79884)
at Module.callPyObject (/Users/lijie.zhang/workspace/pyodide-repo/dist/pyodide.asm.js:9:80266)
at Timeout.wrapper [as _onTimeout] (/Users/lijie.zhang/workspace/pyodide-repo/dist/pyodide.asm.js:9:56949)
at listOnTimeout (node:internal/timers:564:17)
at process.processTimers (node:internal/timers:507:7) {
type: 'RuntimeError',
__error_address: 25425896
}
</details>

**Comment by ryanking13** [comment]
Answered in: https://github.com/pyodide/pyodide/issues/237#issuecomment-1686229421.
